File: 이주영/week2/2577.java
import java.io.*;
import java.util.*;

class Main {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringBuilder sb = new StringBuilder();

    int a = Integer.parseInt(br.readLine());
    int b = Integer.parseInt(br.readLine());
    int c = Integer.parseInt(br.readLine());

    int mul = a * b * c;

    // --- 1 ---
    Map<Integer, Integer> count = new HashMap<>();
    while (mul > 0) {
      int remained = mul % 10;
      mul /= 10;
      count.put(remained, count.getOrDefault(remained, 0) + 1);
    }

    for (int i = 0; i < 10; i++) {
      System.out.println(count.getOrDefault(i, 0));
    }

    // --- 2 ---
    // 배열 사용
    int[] count2 = new int[10];

    while (mul > 0) {
      count2[mul % 10]++;
      mul /= 10;
    }

    for (int i = 0; i < 10; i++) {
      System.out.println(count[i]);
    }

    // --- 3 ---
    // 문자열 사용
    String s = String.valueOf(a * b * c);
    int[] count2 = new int[10];

    for (char c : s.toCharArray()) {
      count2[c - '0']++;
    }

    for (int i = 0; i < 10; i++) {
      System.out.println(count[i]);
    }
  }
}

File: 이주영/week2/2577.py
import sys
input = sys.stdin.readline

a = int(input())
b = int(input())
c = int(input())

mul = a * b * c

# --- 1 ---
count = dict()

while mul > 0:
  temp = mul % 10
  mul = mul // 10
  count[temp] = count.get(temp, 0) + 1

for i in range(10):
  print(count.get(i, 0))

# --- 2 ---
# 배열 사용
count = [0] * 10

while mul > 0:
  count[mul % 10] += 1
  mul //= 10

for i in range(10):
  print(count[i])

# --- 3 ---
# 문자열 사용
count = [0] * 10

for ch in str(mul):
  count[int(ch)] += 1

for i in range(10):
  print(count[i])
